### ISSUE TITLE
UpdateUser not workinkg

### ISSUE THREAD

**Issue by niezle-ziolko** [wp-graphql/wp-graphql#2905]
### Description

Hello,

I have one problem, more specifically I have built a customer panel where the customer has access to downloads and the ability to change their email address and password.

Unfortunately when I try to change the password or email address on the test user account I get a message that I don't have permission to perform such an action.

I have looked in your documentation and in the settings of the plugin itself to see if something can be done about this, but the message keeps popping up and I don't know what else I can do here.

My application is based on GatsbyJS, which is configured with Apollo Client to connect to my Wordpress backend for such things.

My Apollo Client:

`import { ApolloClient, InMemoryCache, HttpLink, ApolloLink, concat } from '@apollo/client';

export const authMiddleware = new ApolloLink((operation, forward) => {
  const session = localStorage.getItem('woo-session');
  if (session) {
    operation.setContext(({ headers: {} }) => ({
      headers: {
        'woocommerce-session': `Session ${session}`,
      }
    }));
  }
  return forward(operation);
});

export const afterware = new ApolloLink((operation, forward) => {
  return forward(operation).map((response) => {
    const context = operation.getContext();
    const { response: { headers } } = context;
    const session = headers.get('woocommerce-session');
    if (session) {
      if (localStorage.getItem('woo-session') !== session) {
        localStorage.setItem('woo-session', headers.get('woocommerce-session'));
      }
    }
    return response;
  });
});

const httpLink = new HttpLink({
  uri: process.env.GRAPH_URL,
});

const client = new ApolloClient({
  link: concat(authMiddleware, httpLink, afterware),
  cache: new InMemoryCache(),
});

export default client;`

My edit data panel:

`import React, { useState } from "react";
import { useMutation } from "@apollo/client";

import { UPDATE_USER_MUTATION } from "../../../query/mutations";
import client from "../../../../apollo/client";

const EditDataTab = ({ customerId }) => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [updateUser] = useMutation(UPDATE_USER_MUTATION, { client }); // Użyj dostarczonego klienta

  const handleSubmit = async (e) => {
    e.preventDefault();

    try {
      await updateUser({
        variables: {
          id: customerId,
          email,
          password,
        },
      });

      console.log("Dane użytkownika zaktualizowane!");
    } catch (error) {
      console.error("Wystąpił błąd podczas aktualizacji danych użytkownika:", error);
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <div>
        <label>
          Nowy adres e-mail:
          <input
            type="email"
            value={email}
            onChange={(e) => setEmail(e.target.value)}
          />
        </label>
      </div>
      <div>
        <label>
          Nowe hasło:
          <input
            type="password"
            value={password}
            onChange={(e) => setPassword(e.target.value)}
          />
        </label>
      </div>
      <button type="submit">Zapisz zmiany</button>
    </form>
  );
};

export default EditDataTab;`

My mutation:

`export const UPDATE_USER_MUTATION = gql`
  mutation UpdateUser($id: ID!, $email: String, $password: String) {
    updateUser(input: { id: $id, email: $email, password: $password }) {
      clientMutationId
    }
  }
`;`

I have tried to get help from WooGraphQL as I am creating an online shop however I have been told by them that they currently have a problem with UpdateCustomer and that I should try to perform this action using UpdateUser.

In the admin panel the action to change password or email address works fine.

### Steps to reproduce

Paste the above code into a GatsbyJS-based application.

### WPGraphQL Version

1.14.10

### WordPress Version

6.3

### PHP Version

8.1.18

### Additional enviornment details

GRAPH_URL=http://localhost/graphql

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have disabled ALL plugins except for WPGraphQL.

- [X] Yes
- [ ] My issue is with compatibility with a specific WordPress plugin, and I have listed all my installed plugins (and version info) above.

**Comment by justlevine** [comment]
Closing as `invalid`. Ref: https://wp-graphql.slack.com/archives/C3NM1M291/p1692343502842249?thread_ts=1692337966.454049&cid=C3NM1M291

Tl;Dr the code above isn't authenticating with WordPress, but rather using a `woocommerce-session` with an `updateUser` mutation instead of the woographql `updateCustomer` mutation.

Bugs with WPGraphQL for WooCommerce should be reported in [that repository](https://github.com/wp-graphql/wp-graphql-woocommerce), and usage questions should be posted as repo `Discussions` or asked in the relevant WPGraphQL Slack channel.
